Implemented support for VK posts database and QML model

antseppa opened the pull request.

chriadam commented on src/lib/abstractimagedownloader.cpp:
```diff
@@ -238,6 +238,7 @@ AbstractImageDownloader::~AbstractImageDownloader()
 void AbstractImageDownloader::queue(const QString &url, const QVariantMap &metadata)
 {
     Q_D(AbstractImageDownloader);
+qWarning() << "AbstractImageDownloader::about to queue image:" << url;
```
probably unintentional

chriadam commented on src/lib/vkimagesdatabase.h:
```diff
@@ -0,0 +1,155 @@
+    bool operator!=(const VKAlbum &other) const { return !(*this == other); }
+};
+
+struct VKImage
```
these structs are quite large, so passing QLists of them around will be inefficient.  Might be worth changing the API to use QVector if we notice poor performance loading large albums.

chriadam replied:
Or alternatively, use the const ptr pattern which is used in some other cases in libsocialcache.

chriadam replied:
Also, I probably made a mistake in not putting these into a namespace, perhaps.  I wonder if they should be public classes within VKImageDatabase, instead of top level classes in the global namespace...

antseppa replied:
That's how they are in VKPostsDatabase. I'll move them inside VKImageDatabase to be consistent.

antseppa replied:
Actually, rather the const ptr pattern to maintain BC as well. Started to do that.

chriadam commented on src/lib/abstractsocialpostcachedatabase.cpp:
```diff
@@ -296,6 +299,32 @@ void AbstractSocialPostCacheDatabase::removePosts(int accountId)
     }
 }
 
+void AbstractSocialPostCacheDatabase::removePost(const QString &identidier)
```
typo: identifier

chriadam commented:
LGTM although in the future we could look into improving the performance of the VK images stuff (probably via the const ptr pattern used elsewhere) and also the class/struct namespacing issues.

antseppa commented:
Updated:
- squashed to one commit
- fixed identifier typo
- removed accidental warning from abstarct image downloader
- changed VKImageDatabase structs to use ConstPtr pattern.
- fixed a bug in VKImageDatabasePrivate::fixed queryUsers() which messed up multi-account case.

chriadam commented:
LGTM.  Not sure that squashing down to one single commit is necessary, but that's fine I guess.

antseppa commented:
Had the original structure still left, decided to unsquash after discussion. Otherwise same.

chriadam commented:
LGTM

antseppa commented:
Updated: rebased

chriadam commented:
Still LGTM
